MediaBrowser.XbmcMetadata/Parsers/EpisodeNfoParser.cs

```diff
@@ -69,6 +73,33 @@ protected override void Fetch(MetadataResult<Episode> item, string metadataFile,
                             }
                         }
                     }
+
+                    // Extract the last episode number from nfo
+                    // This is needed because XBMC metadata uses multiple episodedetails blocks instead of episodenumberend tag
+                    while ((index = xmlFile.IndexOf(srch, StringComparison.OrdinalIgnoreCase)) != -1)
+                    {
+                        xml = xmlFile.Substring(0, index + srch.Length);
+                        xmlFile = xmlFile.Substring(index + srch.Length);
+
+                        using (var stringReader = new StringReader(xml))
+                        using (var reader = XmlReader.Create(stringReader, settings))
+                        {
+                            reader.MoveToContent();
+
+                            if (reader.ReadToDescendant("episode"))
+                            {
+                                var number = reader.ReadElementContentAsString();
+
+                                if (!string.IsNullOrWhiteSpace(number))
```

Comment:
Don't think this is needed since the next line is a try parse and will just fail on an empty/whitespace only string

Reply:
It looks like you are right, I removed it.